commands/up.go

```diff
@@ -67,17 +77,88 @@ func upHandler(cmd *cobra.Command, args []string) error {
 	if err := runBuild(cmd, args); err != nil {
 		return err
 	}
-	fmt.Println()
+
 	if !skipPush {
 		if err := runPush(cmd, args); err != nil {
 			return err
 		}
-		fmt.Println()
 	}
+
 	if !skipDeploy {
 		if err := runDeploy(cmd, args); err != nil {
 			return err
 		}
 	}
+
+	if watch {
+		logger := log.New(os.Stderr, "", log.LstdFlags)
+		logger.SetFlags(log.Ldate | log.Ltime | log.LUTC)
+
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			return err
+		}
+		defer watcher.Close()
+
+		err = filepath.Walk("./", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return watcher.Add(path)
+			}
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		signalChannel := make(chan os.Signal, 1)
+		signal.Notify(signalChannel, os.Interrupt, syscall.SIGTERM)
+
+		go func() {
+			<-signalChannel
+
+			watcher.Close()
+
+			os.Exit(0)
+		}()
```

Comment:
I don't think this exit is needed, after calling close this goroutine will complete and be cleaned up correct. I generally try to avoid calling Exit directly because people often forget where the exit is happening, instead of allowing the function to run to completion. 

Instead, the `for`-loop should properly stop when the `watcher` closes or the signal is submitted.

Looking at this again, this goroutine is not needed at all, this could simply be a `case` in the `switch` 

```go
		for {
			select {
			case event, ok := <-watcher.Events:
				if !ok {
					return fmt.Errorf("watcher's Events channel is closed")
				}

				if event.Op == fsnotify.Write {
					d(func() {
						if err := runBuild(cmd, args); err != nil {
							logger.Fatalf("%s %s", functionName, err)
						}

						if !skipPush {
							if err := runPush(cmd, args); err != nil {
								logger.Fatalf("%s %s", functionName, err)
							}
						}

						if err := runDeploy(cmd, args); err != nil {
							logger.Fatalf("%s %s", functionName, err)
						}
					})
				}

			case err, ok := <-watcher.Errors:
				if !ok {
					return fmt.Errorf("watcher's Errors channel is closed")
				}
				return err
			case <-signalChannel:
				watcher.Close()
				return nil
			}
		}
```

Reply:
@NikhilSharmaWe could you respond to the comments please and click "resolve" if you have resolved the comments?

Reply:
Thanks @LucasRoesler for your reviews